### DISCUSSION TITLE
Client SSH and Telnet Activity

### DISCUSSION THREAD

**Discussion by tm6907** [cmu-sei/GHOSTS#390] [Q&A] [answered]
Hi, Im trying to get a Linux client to run a SSH handler. Logs show it working but theres no evidence of the command actually changing anything on the other VM. 

This is my timeline and credentials file.
![image](https://github.com/user-attachments/assets/005971ee-ece6-433f-b99d-ae2ccfa6bfbe)

![image](https://github.com/user-attachments/assets/1f8cc4b8-56b8-471a-9caa-f7302ac5a4f8)

This is what is shown in the logs before it starts another handler.
![image](https://github.com/user-attachments/assets/edd62b36-e4d3-48d2-a55c-9f24923f3091)

Also wondered is Telnet possible for a client activity?

**Comment by kchen1023** [comment]
Did you convert your password to base64?

**Reply by tm6907** [reply]
Ah my bad, I misunderstood what base64 meant. I'll try it out now

**Reply by tm6907** [reply]
Yeah still not seeing any change from the SSH handler, everything should be setup correctly and theres no indication of any errors.
Same for the SFTP handler I added too.

**Reply by kchen1023** [reply]
Can you post a picture of your password and the logs with the ssh parts?

**Reply by tm6907** [reply]
The logs are the same and for the password I used a base64 encoder to change it to " cGFzc3dvcmQxMjM= "

**Reply by kchen1023** [reply]
Hey, wanted to let you know that I'm currently trying to set up a linux machine. I'll try something similar to yours and see if it works.

**Reply by kchen1023** [reply, marked as answer]
Ok, I just tried it and it works. It seems like it will show in the logs if it is successful.
![image](https://github.com/user-attachments/assets/18f5f3a6-b25a-49c5-87c1-8f02e92e92fb)
![image](https://github.com/user-attachments/assets/f7f98558-1465-428c-bfab-5bd942705eda)

The issue I was having was that it kept saying path not found even though I quadruple checked, but then I realized that the Linux version somehow adds the path to 'build_output' in front of whatever you put in the credentials file.

So instead of
~/Downloads/Ghosts_Client/build_output/ssh_creds.json" for 'CredentialsFile'
which the client will try to go to ~/Downloads/Ghosts_Client/build_output/\~/Downloads/Ghosts_Client/build_output/ssh_creds.json
I changed it to just "ssh_creds.json" and then it worked.
![image](https://github.com/user-attachments/assets/ff15db2a-28b3-4e38-9934-d76037dd5c0b)
